pkg/summarizer/flakiness_test.go

```diff
@@ -0,0 +1,233 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package summarizer
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/testgrid/pb/state"
+	"github.com/GoogleCloudPlatform/testgrid/pkg/summarizer/common"
+)
+
+func TestIsWithinTimeFrame(t *testing.T) {
+	cases := []struct {
+		name      string
+		column    *state.Column
+		startTime int
+		endTime   int
+		expected  bool
+	}{
+		{
+			name: "column within time frame returns true",
+			column: &state.Column{
+				Started: 1.0,
+			},
+			startTime: 0,
+			endTime:   2,
+			expected:  true,
+		},
+		{
+			name: "column outside of time frame returns false",
+			column: &state.Column{
+				Started: 1.0,
+			},
+			startTime: 3,
+			endTime:   7,
+			expected:  false,
+		},
+		{
+			name: "column outside of time frame returns false, test 2",
+			column: &state.Column{
+				Started: 4.0,
+			},
+			startTime: 0,
+			endTime:   2,
+			expected:  false,
+		},
+		{
+			name: "function is inclusive with startTime",
+			column: &state.Column{
+				Started: 0.0,
+			},
+			startTime: 0,
+			endTime:   2,
+			expected:  true,
+		},
+		{
+			name: "function is inclusive with endTime",
+			column: &state.Column{
+				Started: 2.0,
+			},
+			startTime: 0,
+			endTime:   2,
+			expected:  true,
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			if actual := isWithinTimeFrame(tc.column, tc.startTime, tc.endTime); actual != tc.expected {
+				t.Errorf("actual %t != expect %t", actual, tc.expected)
```

Comment:
Error message would probably be more valuable as 
`t.Errorf("isWithinTimeFrame returned %t for %d < %f <= %d", actual, tc.startTime, tc.column.Started, tc.endTime)`

Reply:
Done